File: 235/Recursive.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        if (p.val > q.val) {
            TreeNode temp = q;
            q = p;
            p = temp;
        }
            
        if (p.val <= root.val && root.val <= q.val) return root;
        if (root.val < p.val) return lowestCommonAncestor(root.right, p, q);
        else return lowestCommonAncestor(root.left, p, q);
    }
}

File: 94/recursive.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

def inorderTraversal_rec(root, inOrderNodes):
    if root is None:
        return inOrderNodes
    inOrderNodes = inorderTraversal_rec(root.left, inOrderNodes)
    inOrderNodes.append(root.val)
    inOrderNodes = inorderTraversal_rec(root.right, inOrderNodes)
    return inOrderNodes

class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        return inorderTraversal_rec(root, [])
